src/sys/signal.rs

```diff
@@ -680,6 +680,16 @@ pub fn kill<T: Into<Option<Signal>>>(pid: ::unistd::Pid, signal: T) -> Result<()
     Errno::result(res).map(drop)
 }
 
+pub fn killpg<T: Into<Option<Signal>>>(pid: ::unistd::Pid, signal: T) -> Result<()> {
+    let res = unsafe { libc::killpg(pid.into(),
+                                  match signal.into() {
+                                      Some(s) => s as libc::c_int,
+                                      None => 0,
```

Comment:
Why would you ever want to send 0?  Is that even valid?

Reply:
It's valid. It just perform error checking and won't send any signal.

According to the [POSIX spec of `killpg`](http://pubs.opengroup.org/onlinepubs/9699919799/functions/killpg.html):

> If pgrp is greater than 1, killpg(pgrp, sig) shall be equivalent to kill(-pgrp, sig).

And in the [POSIX spec of `kill`](http://pubs.opengroup.org/onlinepubs/9699919799/functions/kill.html):

> The signal to be sent is specified by sig and is either one from the list given in <signal.h> or 0. If sig is 0 (the null signal), error checking is performed but no signal is actually sent. The null signal can be used to check the validity of pid.

Also in the [manpage of `killpg`](http://man7.org/linux/man-pages/man3/killpg.3.html):

> On Linux, killpg() is implemented as a library function that makes the call kill(-pgrp, sig).